lib/diagnostic.rb

```diff
@@ -5,7 +5,7 @@ class Diagnostic
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    enum.reduce { |memo, obj| memo.length > obj.length ? memo : obj }
```

Comment:
Correct. For your reference, though, Enumerable provides some simpler methods for doing this kind of work: `max` and `max_by`.
e.g.
`enum.max { |a, b| a.length <=> b.length }`
`enum.max_by { |entry| entry.length }`